src/compiler/moduleSpecifiers.ts

```diff
@@ -286,19 +295,16 @@ namespace ts.moduleSpecifiers {
         const redirects = host.redirectTargetsMap.get(importedPath) || emptyArray;
         const importedFileNames = [...(referenceRedirect ? [referenceRedirect] : emptyArray), importedFileName, ...redirects];
         const targets = importedFileNames.map(f => getNormalizedAbsolutePath(f, cwd));
-        let shouldFilterIgnoredPaths = !every(targets, containsIgnoredPath);
+        let shouldFilterIgnoredPaths = host.preferFailureToResultsWithIgnoredPaths?.() || !every(targets, containsIgnoredPath);
 
         if (!preferSymlinks) {
             // Symlinks inside ignored paths are already filtered out of the symlink cache,
             // so we only need to remove them from the realpath filenames.
             const result = forEach(targets, p => !(shouldFilterIgnoredPaths && containsIgnoredPath(p)) && cb(p, referenceRedirect === p));
             if (result) return result;
         }
-        const links = host.getSymlinkCache
-            ? host.getSymlinkCache()
-            : discoverProbableSymlinks(host.getSourceFiles(), getCanonicalFileName, cwd);

```

Comment:
While refactoring for better preservation of a single SymlinkCache instance, I decided to get rid of this fallback. `ModuleSpecifierResolutionHost` is internal and all our implementations of it implement `getSymlinkCache`. It always delegates to `Program.getSymlinkCache` which delegates to `Project.getSymlinkCache`, but Program itself has a fallback. This should never be called directly by public API consumers, and even those who are using internal things should be taken care of via `ts.createModuleSpecifierResolutionHost`, so I think this is safe to remove.